File: lib_lampros/nag17/examples/results/d04aafe.r
 D04AAF Example Program Results
 
 Four separate runs to calculate the first four odd order derivatives of
    FUN(X) = 0.5*exp(2.0*X-1.0) at X = 0.5.
 The exact results are 1, 4, 16 and 64
 
 Input parameters common to all four runs
    XVAL = 0.5    NDER = -7    IFAIL = 0
 
 
 with step length    .5000  the results are
 Order        Derivative       Error estimate
  1            .1392D+04           -.1073D+06
  3           -.3139D+04           -.1438D+06
  5            .8762D+04           -.2479D+06
  7           -.2475D+05           -.4484D+06
 
 with step length    .0500  the results are
 Order        Derivative       Error estimate
  1            .1000D+01            .1530D-10
  3            .4000D+01            .2113D-08
  5            .1600D+02            .3815D-06
  7            .6400D+02            .7385D-04
 
 with step length    .0050  the results are
 Order        Derivative       Error estimate
  1            .1000D+01            .1221D-13
  3            .4000D+01            .4208D-09
  5            .1600D+02            .1450D-04
  7            .6404D+02            .2973D+00
 
 with step length    .0005  the results are
 Order        Derivative       Error estimate
  1            .1000D+01            .1422D-12
  3            .4000D+01            .3087D-06
  5            .1599D+02            .6331D+00
  7            .3825D+05           -.1964D+07

File: lib_lampros/nag17/examples/source/d04aafe.f
*     D04AAF Example Program Text
*     Mark 14 Revised.  NAG Copyright 1989.
*     .. Parameters ..
      INTEGER          NOUT
      PARAMETER        (NOUT=6)
*     .. Local Scalars ..
      DOUBLE PRECISION HBASE, XVAL
      INTEGER          I, IFAIL, J, K, L, NDER
*     .. Local Arrays ..
      DOUBLE PRECISION DER(14), EREST(14)
*     .. External Functions ..
      DOUBLE PRECISION FUN
      EXTERNAL         FUN
*     .. External Subroutines ..
      EXTERNAL         D04AAF
*     .. Intrinsic Functions ..
      INTRINSIC        ABS
*     .. Executable Statements ..
      WRITE (NOUT,*) 'D04AAF Example Program Results'
      WRITE (NOUT,*)
      WRITE (NOUT,*)
     +'Four separate runs to calculate the first four odd order derivati
     +ves of'
      WRITE (NOUT,*) '   FUN(X) = 0.5*exp(2.0*X-1.0) at X = 0.5.'
      WRITE (NOUT,*) 'The exact results are 1, 4, 16 and 64'
      WRITE (NOUT,*)
      WRITE (NOUT,*) 'Input parameters common to all four runs'
      WRITE (NOUT,*) '   XVAL = 0.5    NDER = -7    IFAIL = 0'
      WRITE (NOUT,*)
      HBASE = 0.5D0
      NDER = -7
      L = ABS(NDER)
      IF (NDER.GE.0) THEN
         J = 1
      ELSE
         J = 2
      END IF
      XVAL = 0.5D0
      DO 40 K = 1, 4
         IFAIL = 0
*
         CALL D04AAF(XVAL,NDER,HBASE,DER,EREST,FUN,IFAIL)
*
         WRITE (NOUT,*)
         WRITE (NOUT,99999) 'with step length', HBASE,
     +     '  the results are'
         WRITE (NOUT,*) 'Order        Derivative       Error estimate'
         DO 20 I = 1, L, J
            WRITE (NOUT,99998) I, DER(I), EREST(I)
   20    CONTINUE
         HBASE = HBASE*0.1D0
   40 CONTINUE
      STOP
*
99999 FORMAT (1X,A,F9.4,A)
99998 FORMAT (1X,I2,2D21.4)
      END
*
      DOUBLE PRECISION FUNCTION FUN(X)
*     .. Scalar Arguments ..
      DOUBLE PRECISION              X
*     .. Intrinsic Functions ..
      INTRINSIC                     EXP
*     .. Executable Statements ..
      FUN = 0.5D0*EXP(2.0D0*X-1.0D0)
      RETURN
      END
